src/app/app.component.ts

```diff
@@ -1,10 +1,33 @@
 import { Component } from '@angular/core';
+import { FormGroup, Validators, FormBuilder } from '@angular/forms';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
-  title = 'password-strength';
+  isShowPassword = true;
+  passwordForm: FormGroup;
+  passwordIsValid = false;
+
+  constructor(private fb: FormBuilder) {}
+
+  ngOnInit() {
+    this.passwordForm = this.fb.group({
+      password: ['', Validators.required],
```

Comment:
move all password strength validation logic into [custom validator](https://angular.io/guide/form-validation#defining-custom-validators) so always have it in one place and can check `control.valid`